models/ROMS_rutgers/model_mod.f90

```diff
@@ -0,0 +1,1716 @@
+! to be conveyed to ROMS to dictate the length of the forecast.
+! This file is then used by scripts to modify the ROMS run.
+! The format in the time information is totally at your discretion.
+!
+! ncid:        File id
+! model_time:  The current time of the model state
+! adv_to_time: The time in the future of the next assimilation.
+
+subroutine write_model_time(ncid, model_time, adv_to_time)
+
+character(len=*), parameter :: routine = 'write_model_time'
+
+integer,         intent(in)           :: ncid
+type(time_type), intent(in)           :: model_time
+type(time_type), intent(in), optional :: adv_to_time
+
+integer         :: io, varid, seconds, days
+type(time_type) :: origin_time, deltatime
+real(digits12)  :: run_duration
+
+if (.not. module_initialized) call static_init_model
+
+if (present(adv_to_time)) then
+   string3 = time_to_string(adv_to_time)
+   
+   write(string1, *) 'ROMS/DART not configured to advance ROMS.'
+   write(string2, *) 'called with optional advance_to_time of'
+   
+   call error_handler(E_ERR, routine, string1, &
+              source, text2=string2,text3=string3)
+endif
+
+! If the ocean_time variable exists, we are updating a ROMS file,
+! if not ... must be updating a DART diagnostic file.
+
+io = nf90_inq_varid(ncid, 'ocean_time', varid)
+if (io == NF90_NOERR) then
+   call get_time_information('unknown', ncid, 'ocean_time', 'ocean_time', &
+                myvarid=varid, origin_time=origin_time)
+   deltatime = model_time - origin_time
+   call get_time(deltatime, seconds, days)
+   run_duration = real(days,digits12)*86400.0_digits12 + real(seconds,digits12)
+   call nc_check(nf90_put_var(ncid, varid, run_duration), routine, 'put_var')
+   return
+endif
+
+io = nf90_inq_varid(ncid, 'time', varid)
+if (io == NF90_NOERR) then
+   call get_time_information('unknown', ncid, 'time', 'time', &
+                myvarid=varid, origin_time=origin_time)
+   deltatime = model_time - origin_time
+   call get_time(deltatime, seconds, days)
+   run_duration = real(days,digits12)*86400.0_digits12 + real(seconds,digits12)
+   call nc_check(nf90_put_var(ncid, varid, run_duration), routine, 'put_var')
+   return
+endif
+
+end subroutine write_model_time
+
+
+!--------------------------------------------------------------------
+! Read the time from the input file
+! filename: Name of file that contains the time
+
+function read_model_time(filename)
+
+character(len=*), parameter :: routine = 'read_model_time'
+
+character(len=*), intent(in) :: filename
+type(time_type)              :: read_model_time
+
+integer :: ncid
+
+if (.not. module_initialized) call static_init_model
+
+if (.not. file_exist(filename)) then
+   write(string1,*) 'cannot open file ', trim(filename),' for reading.'
+   call error_handler(E_ERR, routine, string1, source)
+endif
+
+ncid = nc_open_file_readonly(trim(filename), routine)
+
+call get_time_information(filename, ncid, 'ocean_time', 'ocean_time', last_time=read_model_time)
```

Comment:
This doesn't have a matching check for 'time' vs 'ocean' time but I think only the first ensemble member gets its time read. 

https://github.com/NCAR/DART/blob/f7205cb8bb947d2eb3580e5e81e77e5720430868/assimilation_code/modules/io/direct_netcdf_mod.f90#L515-L517